src/operator/subgraph/mkldnn/mkldnn_conv.cc

```diff
@@ -0,0 +1,325 @@
+    if (it != attrs.dict.end())
+      in_sum_at_begin = (it->second == "true");
+    it = attrs.dict.find("with_bn");
+    if (it != attrs.dict.end())
+      with_bn = (it->second == "true");
+    it = attrs.dict.find("with_relu");
+    if (it != attrs.dict.end())
+      with_relu = (it->second == "true");
+    it = attrs.dict.find("with_sum");
+    if (it != attrs.dict.end())
+      with_sum = (it->second == "true");
+    it = attrs.dict.find("with_postsum_relu");
+    if (it != attrs.dict.end())
+      with_postsum_relu = (it->second == "true");
+
+    DFSVisit(subgraph_sym_.outputs, [&](const nnvm::NodePtr &node) {
+      if (node->is_variable()) return;
+      auto &node_name = node->op()->name;
+      if (node_name == "BatchNorm") {
+        CHECK(bn_attrs_.get() == nullptr);
+        CHECK_EQ(with_bn, true);
+        bn_attrs_ = std::make_shared<nnvm::NodeAttrs>(node->attrs);
+      } else if (node_name == "Convolution") {
+        CHECK(conv_attrs_.get() == nullptr);
+        conv_attrs_ = std::make_shared<nnvm::NodeAttrs>(node->attrs);
+      }
+    });
+    CHECK(conv_attrs_.get());
+    conv_attrs_->dict["with_bn"] = with_bn ? "true" : "false";
+    conv_attrs_->dict["with_relu"] = with_relu ? "true" : "false";
+    conv_attrs_->dict["with_sum"] = with_sum ? "true" : "false";
+    conv_attrs_->dict["with_postsum_relu"] = with_postsum_relu ? "true" : "false";
+  }
+
+  void Forward(const OpContext &ctx,
+               const std::vector<NDArray> &inputs,
+               const std::vector<OpReqType> &req,
+               const std::vector<NDArray> &outputs);
+
+  void Backward(const OpContext &ctx,
+                const std::vector<NDArray> &inputs,
+                const std::vector<OpReqType> &req,
+                const std::vector<NDArray> &outputs) {
+    LOG(FATAL) << "Not implemented: subgraph mkldnn Conv only supports inference computation";
+  }
+
+ private:
+  nnvm::Symbol subgraph_sym_;
+  // CachedOpPtr subgraph_exec_; // Used for fallback compute
+  std::shared_ptr<NDArray> cached_weight_;
+  std::shared_ptr<NDArray> cached_bias_;
+  std::shared_ptr<nnvm::NodeAttrs> bn_attrs_;
+  std::shared_ptr<nnvm::NodeAttrs> conv_attrs_;
+  bool in_sum_at_begin;
+  bool with_bn;
+  bool with_relu;
+  bool with_sum;
+  bool with_postsum_relu;
+};
+
+void SgMKLDNNConvOperator::Forward(const OpContext &ctx,
+                                   const std::vector<NDArray> &inputs,
+                                   const std::vector<OpReqType> &req,
+                                   const std::vector<NDArray> &outputs) {
+  const ConvolutionParam &conv_params = nnvm::get<ConvolutionParam>(conv_attrs_->parsed);
+#if SUBGRAPH_DEBUG
+  LOG(INFO) << "Conv inputs size: " << inputs.size();
+  LOG(INFO) << "Conv outputs size: " << outputs.size();
+  LOG(INFO) << "Conv req size: " << req.size();
+  for (size_t k = 0; k < inputs.size(); ++k) {
+    auto input = inputs[k];
+    printf("input %ld :", k);
```

Comment:
Sometimes it's LOG(INFO), sometimes it's printf. I think it's better to use LOG always.